src/Datadog.Trace/FrameworkDescription.NetCore.cs

```diff
@@ -8,43 +8,64 @@ namespace Datadog.Trace
 {
     internal partial class FrameworkDescription
     {
-        public static FrameworkDescription Create()
+        private static FrameworkDescription _instance = null;
+
+        public static FrameworkDescription Instance
+        {
+            get { return _instance ?? (_instance = Create()); }
+        }
+
+        private static FrameworkDescription Create()
         {
-            string frameworkName = null;
-            string osPlatform = null;
+            var frameworkName = "unknown";
+            var frameworkDescription = "unknown";
+            var osPlatform = "unknown";
+            var osArchitecture = "unknown";
+            var processArchitecture = "unknown";
 
             try
             {
-                // RuntimeInformation.FrameworkDescription returns a string like ".NET Framework 4.7.2" or ".NET Core 2.1",
-                // we want to return everything before the last space
-                string frameworkDescription = RuntimeInformation.FrameworkDescription;
-                int index = frameworkDescription.LastIndexOf(' ');
-                frameworkName = frameworkDescription.Substring(0, index).Trim();
-            }
-            catch (Exception e)
-            {
-                Log.SafeLogError(e, "Error getting framework name from RuntimeInformation");
-            }
+                try
+                {
+                    // RuntimeInformation.FrameworkDescription returns a string like ".NET Framework 4.7.2" or ".NET Core 2.1",
+                    // we want to return everything before the last space
+                    frameworkDescription = RuntimeInformation.FrameworkDescription;
```

Comment:
nit: Can this variable also be renamed? Maybe `frameworkVersion` since we end up using this as the version.

Reply:
Done